lib/f32.fz

```diff
@@ -149,12 +169,14 @@ f32(val f32) : float is
 
   # equality
   #
-  fixed type.equality(a, b f32) bool is intrinsic
+  fixed type.equality(a, b f32) bool is
+    a.infix = b || (is_NaN a && is_NaN b)
 
 
   # total order
```

Comment:
Can you please add a comment that this provides a total order in the mathematical sense and therefore breaks the IEEE-semantics. `infix <=` etc.  should be used for IEEEE semantics.